src/mesh/replicated_mesh.C

```diff
@@ -173,6 +173,37 @@ ReplicatedMesh::ReplicatedMesh (const UnstructuredMesh & other_mesh) :
       other_boundary_info.get_nodeset_name(node_bnd_id);
 }
 
+ReplicatedMesh & ReplicatedMesh::operator= (ReplicatedMesh && other_mesh)
+{
+  // First make a call to MeshBase::clear
+  this->MeshBase::clear();
+
+  // Move assign as a ParallelObject.
+  this->ParallelObject::operator=(other_mesh);
+
+  // Move assign as a MeshBase.
+  this->MeshBase::operator=(std::move(other_mesh));
+
+  // Nodes and elements belong to ReplicatedMesh and have to be
+  // moved before we can move the BoundaryInfo object in UnstructuredMesh::operator=.
+  this->move_nodes_and_elements(std::move(other_mesh));
+
+  // Move assign as an UnstructuredMesh.
+  this->UnstructuredMesh::operator=(std::move(other_mesh));
+
+  return *this;
+}
+
+
+void ReplicatedMesh::move_nodes_and_elements(MeshBase && other_mesh)
+{
+  this->_nodes = std::move((libmesh_cast_ptr<ReplicatedMesh*>(&other_mesh))->_nodes);
+  this->_n_nodes = dof_id_type(_nodes.size());
```

Comment:
This isn't nearly as inaccurate, but it's still inaccurate.  We keep _n_nodes so we can keep track of nodes which have been nulled out but (because the container is indexed by id and because vector entry removal is O(N)) haven't been removed from the vector.